pkg/entities/guild.go

```diff
@@ -241,3 +241,92 @@ func (e *Entity) GetGuildRoles(guildID string) (*response.DiscordGuildRoles, err
 
 	return &resp, nil
 }
+
+func (e *Entity) FetchAndSyncGuilds() ([]*response.GetGuildResponse, error) {
+	res, err := e.GetGuilds()
+	if err != nil {
+		e.log.Error(err, "[entity.FetchAndSyncGuilds] GetGuilds() failed")
+		return nil, err
+	}
+
+	dbGuilds := make(map[string]*response.GetGuildResponse)
+	for _, g := range res.Data {
+		dbGuilds[g.ID] = g
+	}
+
+	fetchedGuilds := make(map[string]*discordgo.Guild)
+	e.discord.StateEnabled = true
+	fmt.Println("e.discord.State.Guilds:", e.discord.State.Guilds)
+	for _, g := range e.discord.State.Guilds {
+		fetchedGuilds[g.ID] = g
+	}
+
+	result := make([]*response.GetGuildResponse, 0)
+
+	for id, g := range dbGuilds {
+		_, ok := fetchedGuilds[id]
+		if !ok {
```

Comment:
nhìn nó bằng phẳng hơn thôi chứ kcg
```suggestion
		if ok {
		   result = append(result, g)
		   continue 
		}
		bla bla
```